src/completers/completeTypes/index.js

```diff
@@ -1,21 +1,32 @@
 import { isStringArray } from '../../utils/typeUtils';
 
-function completeTypes(types, ignoredActions = []) {
-  if (!isStringArray(types)) {
-    throw new Error('Types must be an array of strings');
+export const primaryActionsCompleter = primaryActions => primaryActions.reduce(
+  (completedTypes, type) => [...completedTypes, type, `${type}_SUCCESS`, `${type}_FAILURE`],
+  []
+);
+
+export const ignoredActionsCompleter = ignoredActions => ignoredActions || [];
+
+function completeTypes({ primaryActions, ignoredActions = [], customCompleters = [] }) {
+  if (!isStringArray(primaryActions)) {
+    throw new Error('Primary actions must be an array of strings');
   }
   if (!isStringArray(ignoredActions)) {
-    throw new Error('Exception cases from actions must be an array of strings');
+    throw new Error('Ignored actions must be an array of strings');
   }
-
-  const completedTypes = [];
-  types.forEach(type => {
-    completedTypes.push(type);
-    completedTypes.push(`${type}_SUCCESS`);
-    completedTypes.push(`${type}_FAILURE`);
+  customCompleters.forEach(({ actions, completer }) => {
+    if (!isStringArray(actions)) throw new Error('Exception cases from actions must be an array of strings');
+    if (typeof completer !== 'function') throw new Error('Completer must be a function');
   });
 
-  return [...completedTypes, ...ignoredActions];
+  const primaryTypes = primaryActionsCompleter(primaryActions);
+  const ignoredTypes = ignoredActionsCompleter(ignoredActions);
+  const customCompletedTypes = customCompleters.reduce(
+    (customTypes, { actions, completer }) => [...customTypes, ...actions.flatMap(action => completer(action))],
+    []
+  );
+
```

Comment:
I'd add a validation here to check that `customCompletedTypes` is an array of strings